src/Nethermind/Nethermind.Blockchain.Test/TransactionSelectorTests.cs

```diff
@@ -183,6 +184,41 @@ public static IEnumerable EnoughShardBlobTransactionsSelectedTestCases
                 yield return new TestCaseData(enoughTransactionsSelected).SetName(
                     "Enough shard blob transactions and others selected");
+
+                ProperTransactionsSelectedTestCase higherPriorityTransactionsSelected = ProperTransactionsSelectedTestCase.Eip1559Default;
+                var accounts = higherPriorityTransactionsSelected.AccountStates;
+                accounts[TestItem.AddressA] = (1000, 0);
+                accounts[TestItem.AddressB] = (1000, 0);
+                accounts[TestItem.AddressC] = (1000, 0);
+                accounts[TestItem.AddressD] = (1000, 0);
+                accounts[TestItem.AddressE] = (1000, 0);
+                accounts[TestItem.AddressF] = (1000, 0);
+                higherPriorityTransactionsSelected.ReleaseSpec = Cancun.Instance;
+                higherPriorityTransactionsSelected.BaseFee = 1;
+                higherPriorityTransactionsSelected.Transactions = new List<Transaction>
+                {
+                    Build.A.Transaction.WithSenderAddress(TestItem.AddressA).WithType(TxType.Blob).WithNonce(1)
+                        .WithMaxFeePerGas(20).WithMaxFeePerBlobGas(1).WithGasLimit(20).WithBlobVersionedHashes([[0],[0],[0],[0],[0]]).SignedAndResolved(TestItem.PrivateKeyA).TestObject,
+                    Build.A.Transaction.WithSenderAddress(TestItem.AddressB).WithType(TxType.Blob).WithNonce(1)
+                        .WithMaxFeePerGas(16).WithMaxFeePerBlobGas(1).WithGasLimit(20).WithBlobVersionedHashes([[0]]).SignedAndResolved(TestItem.PrivateKeyB).TestObject,
+                    Build.A.Transaction.WithSenderAddress(TestItem.AddressC).WithType(TxType.Blob).WithNonce(1)
+                        .WithMaxFeePerGas(18).WithMaxFeePerBlobGas(1).WithGasLimit(20).WithBlobVersionedHashes([[0]]).SignedAndResolved(TestItem.PrivateKeyC).TestObject,
+                    Build.A.Transaction.WithSenderAddress(TestItem.AddressD).WithType(TxType.Blob).WithNonce(1)
+                        .WithMaxFeePerGas(17).WithMaxFeePerBlobGas(1).WithGasLimit(20).WithBlobVersionedHashes([[0]]).SignedAndResolved(TestItem.PrivateKeyD).TestObject,
+                    Build.A.Transaction.WithSenderAddress(TestItem.AddressE).WithType(TxType.Blob).WithNonce(1)
+                        .WithMaxFeePerGas(19).WithMaxFeePerBlobGas(1).WithGasLimit(20).WithBlobVersionedHashes([[0]]).SignedAndResolved(TestItem.PrivateKeyE).TestObject,
+                    Build.A.Transaction.WithSenderAddress(TestItem.AddressF).WithType(TxType.Blob).WithNonce(1)
+                        .WithMaxFeePerGas(20).WithMaxFeePerBlobGas(1).WithGasLimit(20).WithBlobVersionedHashes([[0]]).SignedAndResolved(TestItem.PrivateKeyF).TestObject,
+                };
+
+                higherPriorityTransactionsSelected.ExpectedSelectedTransactions.AddRange(
+                    higherPriorityTransactionsSelected.Transactions.Where(tx => tx.GetBlobCount() == 1)
```

Comment:
My understanding is that you omit the biggest one as the others provide the same fee without occupying 5 blobs.

I'd add a test with more variety in ranges to select from. Why not differentiate on the max price, ordering and more. Also, this could be tested in isolation.

Reply:
Random iterations changes submission ordering (as there are different paths based on how many single blob txs come first)

Other 5 provide higher total fee; isn't how many blobs they use as that's irrelevant to the producer; however the blob cap means can't have the 5 blob tx and the others